assets/src/edit-story/components/inspector/document/publish/publishTime.js

```diff
@@ -89,9 +89,14 @@ function PublishTime() {
       updateStory,
     })
   );
-  const { timeFormat } = useConfig();
+  const {
+    locale: { timeFormat },
+  } = useConfig();
   const use12HourFormat = is12Hour(timeFormat);
```

Comment:
This function (`is12hour`) is only ever used here - if we just make it stateful as well (depending on the settings inside `date` package), we can remove this dependency on config here.

And then we can completely remove `locale` from `config` altogether, as this appears to be the only instance where anything from the `locale` property is used through `useConfig()`.